numpy/lib/function_base.py

```diff
@@ -3612,17 +3786,81 @@ def _quantile_is_valid(q):
     return True
 
 
-def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
-                           interpolation='linear', keepdims=False):
+def _validate_weights(a, q, axis, weights):
+    if weights is None:
+        wgt = None
+    else:
+        a = asanyarray(a)
+        wgt = np.asanyarray(weights)
+
+        if issubclass(a.dtype.type, (np.integer, np.bool_)):
```

Comment:
I don't think boolean weights should be supported. `isinstance(np.bool_, np.number)` is deliberately false

Reply:
Will fix